lib/media.js

```diff
@@ -1259,21 +1265,31 @@ module.exports = function(Cam) {
 		this._request({
 			service: mediaType
 			, body: this._envelopeHeader() +
-				`<wsdl:CreateOSD xmlns:wsdl="${mediaNS}" xmlns:sch="http://www.onvif.org/ver10/schema">
+              `<wsdl:CreateOSD xmlns:wsdl="${mediaNS}" xmlns:sch="http://www.onvif.org/ver10/schema">
          <wsdl:OSD token="">
-            <sch:VideoSourceConfigurationToken>${(options.videoSourceConfiguationToken || this.activeSource.videoSourceConfigurationToken)}</sch:VideoSourceConfigurationToken>
-            <sch:Type>Text</sch:Type>
-            <sch:Position>
-               <sch:Type>${options.position || 'LowerLeft'}</sch:Type>
-            </sch:Position>
-            <sch:TextString IsPersistentText="false">
-               <sch:Type>Plain</sch:Type>
-               <sch:PlainText>${options.plaintext}</sch:PlainText>
-            </sch:TextString>
+            <sch:VideoSourceConfigurationToken>${options.videoSourceConfiguationToken || this.activeSource.videoSourceConfigurationToken}</sch:VideoSourceConfigurationToken>
+						<sch:Type>Text</sch:Type>
+						<sch:Position>
+							<sch:Type>${ typeof options.position === "object" ? "Custom" : options.position ? options.position : "LowerLeft"}</sch:Type>
+				   ${typeof options.position === "object" ? '<sch:Pos x="' + options.position.x + '" y="' + options.position.y + '"/>' : ""}
+				</sch:Position>
+				  <sch:TextString IsPersistentText="false">
+					${ options.dateFormat && options.timeFormat ? 
+		`<sch:Type>DateAndTime</sch:Type>
+									<sch:DateFormat>${options.dateFormat}</sch:DateFormat>
+									<sch:TimeFormat>${options.timeFormat}</sch:TimeFormat>`
+		: `<sch:Type>Plain</sch:Type>
+									<sch:PlainText>${options.plaintext}</sch:PlainText>`}
+
+					${options.fontSize ? `<sch:FontSize>${options.fontSize}</sch:FontSize>` : ""}				
+					${options.fontColor ? `
+						<sch:FontColor>
+				        ${'<sch:Color Z="' + options.fontColor.Z + '" Y="' + options.fontColor.Y + '" X="' + options.fontColor.X + '"/>'}
```

Comment:
Maybe add `Colorspace="http://www.onvif.org/ver10/colorspace/RGB"` here? Or add optional param to choose between YCbCr and RGB?

Reply:
Added new options parameter - colorspace.
Supporting YCbCr and RGB.